narwhals/dtypes.py

```diff
@@ -571,14 +654,35 @@ class Field:
     """
 
     name: str
+    """The name of the field within its parent `Struct`."""
     dtype: IntoDType
+    """The `DType` of the field's values."""
 
     def __init__(self, name: str, dtype: IntoDType) -> None:
         self.name = name
         self.dtype = dtype
 
     def __eq__(self, other: Field) -> bool:  # type: ignore[override]
-        return (self.name == other.name) & (self.dtype == other.dtype)
+        """Check if this Field is equivalent to another Field.
+
```

Comment:
Examples are helpful and users will extrapolate easily, but a direct statement would also be nice. Something like:

```suggestion
        """Check if this Field is equivalent to another Field.

        Two fields are equivalent if they have the same name and the same dtype.
```